src/databricks/labs/remorph/reconcile/query_builder/count_query.py

```diff
@@ -0,0 +1,33 @@
+import logging
+
+from sqlglot import Dialect
+from sqlglot import expressions as exp
+
+from databricks.labs.remorph.reconcile.query_builder.expression_generator import build_column, build_literal
+from databricks.labs.remorph.reconcile.recon_config import Table
+
+logger = logging.getLogger(__name__)
+
+
+class CountQueryBuilder:
```

Comment:
Way too many abstraction, @bishwajit-db any thoughts from you on how we can simplify this?

Reply:
It requires a lot of changes. will do it later.

Reply:
As discussed this will require a re-design taking it as tech debt.